### PR TITLE
match dh and hcris facility data

### PR DESCRIPTION
**Pull request by simonkassel**
Better joining of HCRIS data with DH.

Start by re-geocoding DH data and checking google geocoding results against build-in DH points (the former are better). Then match points through a combination

Adresses #70 

TO DO:
- [x] Geocode DH 
- [x] Clean up errors in DH geocoding (fallback to original DH points where applicable)
- [x] Nearest neighbor matching between datasets
- [x] Fuzzy matching for ambigous cases
- [x] Folium map for spotchecking joining results
- [ ] Re-factor geocoding code
- [ ] Updated DH data
- [ ] Apply process to HIFLD
- [ ] Documentation

### CODE REVIEW

**Comment by simonkassel** [comment]
Still need to finish up several tasks (most notably adding HIFLD and new DH data) but we're going to try to get the latest data set version merged now and address those other items in the next PR

File: notebooks/processing/03_Merge_Facility_Information.ipynb [hunk 1]
```diff
@@ -6,164 +6,350 @@
...
     "# Merge facility information\n",
     "\n",
-    "Merge facility data from HCRIS (Healthcare Cost Reporting Information System) andDH (Definitive Healthcare) datasets."
+    "Merge facility data from HCRIS (Healthcare Cost Reporting Information System) and DH (Definitive Healthcare) datasets."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 59,
```
Comment (lossyrob) [inline notebooks/processing/03_Merge_Facility_Information.ipynb:14]: Uncleared notebook

File: notebooks/processing/02_Process_HCRIS_Data.ipynb [hunk 2]
```diff
@@ -754,13 +754,6 @@
     "with open(processed_data_path('usa_hospital_beds_hcris2018.geojson'), 'w') as f:\n",
     "    f.write(json.dumps(geojson, indent=4))"
    ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": []
```
Comment (lossyrob) [inline notebooks/processing/02_Process_HCRIS_Data.ipynb:763]: Seems like an extra cell got added here.

File: notebooks/processing/03_Merge_Facility_Information.ipynb [hunk 3]
```diff
@@ -6,164 +6,350 @@
...
-    "                        rsimilarity_properties = similarity_columns['HCRIS'],\n",
-    "                        similarity_weights=[0.6, 0.4],                                                       \n",
-    "                        distance=1000,\n",
-    "                        merge_unmatched=False)"
+    "hcris_to_dh_matches, hcris_to_dh_data = match_by_state(\n",
+    "    hcris_info, \n",
+    "    dh_info, \n",
+    "    'state_validation_maps_03-31-21_hcris-to-dh')\n",
+    "\n",
+    "HtoD_matches = hcris_to_dh_data['matching_dfs'].as_type(str)\n",
```
[the remark this answers is not in the document]
Comment (simonkassel) [reply notebooks/processing/03_Merge_Facility_Information.ipynb:265]: got it

File: notebooks/processing/03_Merge_Facility_Information.ipynb (hunk 1 above)
Comment (simonkassel) [reply notebooks/processing/03_Merge_Facility_Information.ipynb:14]: ok, will clear

**Comment by simonkassel** [comment]
@lossyrob I added documentation to the matching code. I also made a very minor change in that process and now there are a few more matches so there is an updated dataset in there too

**Comment by lossyrob** [comment]
I used this PR as a base for #120, which enables using HIFLD data. Closing this, the commits will be merged with #120.

**Pull request closed without merging**